components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -268,25 +320,38 @@ def compress(
         task_id,
         tag_ids,
         paths_to_compress,
+        sql_adapter,
         clp_metadata_db_connection_config,
     )
     duration = (datetime.datetime.now() - start_time).total_seconds()
     logger.info(f"[job_id={job_id} task_id={task_id}] COMPRESSION COMPLETED.")
 
-    if compression_successful:
-        return CompressionTaskSuccessResult(
-            task_id=task_id,
-            status=CompressionTaskStatus.SUCCEEDED,
-            start_time=start_time,
-            duration=duration,
-            total_uncompressed_size=worker_output["total_uncompressed_size"],
-            total_compressed_size=worker_output["total_compressed_size"],
-        ).dict()
-    else:
-        return CompressionTaskFailureResult(
+    with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+        db_conn.cursor(dictionary=True)
+    ) as db_cursor:
+        update_compression_task_metadata(
+            db_cursor,
+            task_id,
+            dict(
+                start_time=start_time,
+                status=compression_task_status,
+                partition_uncompressed_size=worker_output["total_uncompressed_size"],
+                partition_compressed_size=worker_output["total_compressed_size"],
+                duration=duration,
+            ),
+        )
+        db_conn.commit()
+
+        compression_task_result = CompressionTaskResult(
             task_id=task_id,
-            status=CompressionTaskStatus.FAILED,
-            start_time=start_time,
+            status=compression_task_status,
             duration=duration,
-            error_message=worker_output["error_message"],
-        ).dict()
+        )
+
+        if CompressionTaskStatus.SUCCEEDED == compression_task_status:
+            increment_compression_job_metadata(db_cursor, job_id, dict(num_tasks_completed=1))
+            db_conn.commit()
```

Comment:
Instead of separating the database ops, could we do all the database ops first and then prepare the CompressionTaskResult afterward, outside the `with` block? Something like:

```py
with ...:
  update_compression_task_metadata(...)

  if CompressionTaskStatus.SUCCEEDED == compression_task_status:
    increment_compression_job_metadata(...)

  db_conn.commit()

compression_task_result = ...
if CompressionTaskStatus.SUCCEEDED != compression_task_status:
  compression_task_result.error_message = ...
```

That would require repeating the `CompressionTaskStatus.SUCCEEDED` check, but I think it's a little clearer in terms of grouping operations.